devicetypes/smartthings/zll-rgb-bulb.src/zll-rgb-bulb.groovy

```diff
@@ -73,14 +98,33 @@ def parse(String description) {
 		log.trace "zigbeeMap : $zigbeeMap"
 
 		if (zigbeeMap?.clusterInt == COLOR_CONTROL_CLUSTER) {
-			if(zigbeeMap.attrInt == ATTRIBUTE_HUE){  //Hue Attribute
+			if(zigbeeMap.attrInt == ATTRIBUTE_HUE && shouldUseHueSaturation()){  //Hue Attribute
 				def hueValue = Math.round(zigbee.convertHexToInt(zigbeeMap.value) / 0xfe * 100)
 				sendEvent(name: "hue", value: hueValue, displayed:false)
 			}
-			else if(zigbeeMap.attrInt == ATTRIBUTE_SATURATION){ //Saturation Attribute
+			else if(zigbeeMap.attrInt == ATTRIBUTE_SATURATION && shouldUseHueSaturation()) { //Saturation Attribute
 				def saturationValue = Math.round(zigbee.convertHexToInt(zigbeeMap.value) / 0xfe * 100)
 				sendEvent(name: "saturation", value: saturationValue, displayed:false)
 			}
+			else if(zigbeeMap.attrInt == ATTRIBUTE_X) { //X Attribute
+				def xValue = zigbee.convertHexToInt(zigbeeMap.value)
+				log.debug "xValue = $xValue"
+
+				state.currentX = xValue
+			}
+			else if(zigbeeMap.attrInt == ATTRIBUTE_Y) { //Y Attribute
+				def yValue = zigbee.convertHexToInt(zigbeeMap.value)
+				log.debug "yValue = $yValue"
+
+				state.currentY = yValue
+			}
+
+			if (!shouldUseHueSaturation() && state.currentX && state.currentY) {
+				def hsv = safeColorXy2Hsv(state.currentX, state.currentY)
```

Comment:
Did you do testing of this on the cloud side.  I was working on this locally and the hue/sat values that we get out don't really match up with the colors.